### DISCUSSION TITLE
Sysex Messages chrash the browser

### DISCUSSION THREAD

**Discussion by schoko11** [djipco/webmidi#334] [Q&A]
Hi, thanks for this nice project, but i have problems with sysex messages...

I am using vscodium(on WIN10) and Live Server extension(https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) to develop a midi controller for the browser(https://github.com/schoko11/KEKO-WEB).
For midi i use a bluetooth midi device (WIDI Master pro) from CME (to control a kemper amp).

When launching the site with Live Server(in vscodium), the browser gets closed after receiving a few sysex messages(turning the knobs on the hardware).
Chrome debug logs shows nothing but the sysex messages.

The log files in vscodium don't show any hint...

The code is fairly simply, maybe just an silly error somewhere ?
```
import {Webmidi} from xxxx;

const kemperMidi = WebMidi.enable({sysex: true}).then(onEnabled);
let kemperMidiOut;
let kemperMidiIn;

function onEnabled() {
    //Object.keys(obj);
    WebMidi.inputs.forEach(input => console.log(input.manufacturer + "#" + input.name));
    WebMidi.outputs.forEach(output => console.log(output.manufacturer + "#" + output.name));
    kemperMidiOut = WebMidi.getOutputByName("WIDI Master OUT");
    kemperMidiIn = WebMidi.getInputByName("WIDI Master IN");
    console.log(kemperMidiOut);
    debugger;
    kemperMidiIn.addListener("sysex", (e) => {
        //debugger;
        console.log("sysex in:" + Object.keys(e));
        console.log("#" + e.message + "##" + e.rawData + "###" + e.type + "####" + e.dataBytes + "#####" + e.statusByte);
    });
```

Didn't found a reason for the browser crashes (chrome 109 on WIN 10).

Any hints ? 

Thanks!

**Comment by djipco** [comment]
Since there are many parts involved in your question, the best thing to do would be to try and isolate the source of the problem. Do you get the same crash when using vanilla Chrome (no vscodium, no Live Server Extension) ?

**Reply by schoko11** [reply]
thanks for getting back to it...

i tried to open the index.html simply in the browser and got a CORS error, maybe that was the problem...(i think i had to supply the correct header - i read that somewhere)

do you mean this?
https://www.bromite.org/chromium

**Comment by djipco** [comment]
> i tried to open the index.html simply in the browser and got a CORS error, maybe that was the problem...(i think i had to supply the correct header - i read that somewhere)

CORS is something else. You need to load the page through a server.

> do you mean this? https://www.bromite.org/chromium

No, I just meant without any special environment such as vscodium or Live Server extension.

**Reply by schoko11** [reply]
Well the live server is just doing that ???

Tried it now as IIFE too(instead of usage as module) , just to get somehow closer to the problem...

When trying with firefox, i got a crash report...
```
AdapterDeviceID: 0x1f82
AdapterDriverVersion: 30.0.14.9676
AdapterSubsysID: 00000000
AdapterVendorID: 0x10de
....
DOMFissionEnabled: 1
DOMIPCEnabled: 1
EMCheckCompatibility: true
ExperimentalFeatures: devtools.inspector.compatibility.enabled
GPUProcessLaunchCount: 1
GPUProcessStatus: Running
GpuSandboxLevel: 0
HeadlessMode: 0
InstallTime: 1674914027
LauncherProcessState: 0
ModuleSignatureInfo: {"KORG INC.":["KORGBM64.DRV"],"Microsoft Corporation":
["msvcp140.dll","VCRUNTIME140_1.dll","VCRUNTIME140.dll"]...
.....
Vendor: Mozilla
Version: 109.0
Winsock_LSP: MSAFD-Tcpip [TCP/IP] : 2 : 2 : 1 : 6 : 0x20066 : 0x8 : %SystemRoot%\system32\mswsock.dll :  : e70f1aa0-ab8b-11cf-8ca3-00805f48a192 
 MSAFD-Tcpip [UDP/IP] : 2 : 2 : 2 : 17 : 0x20609 : 0x8 : %SystemRoot%\system32\mswsock.dll :  : e70f1aa0-ab8b-11cf-8ca3-00805f48a192 
 MSAFD-Tcpip [RAW/IP] : 2 : 2 : 3 : 0 : 0x20609 : 0xc : %SystemRoot%\system32\mswsock.dll :  : e70f1aa0-ab8b-11cf-8ca3-00805f48a192 
 MSAFD-Tcpip [TCP/IPv6] : 2 : 23 : 1 : 6 : 0x20066 : 0x8 : %SystemRoot%\system32\mswsock.dll :  : f9eab0c0-26d4-11d0-bbbf-00aa006c34e4 
 MSAFD-Tcpip [UDP/IPv6] : 2 : 23 : 2 : 17 : 0x20609 : 0x8 : %SystemRoot%\system32\mswsock.dll :  : f9eab0c0-26d4-11d0-bbbf-00aa006c34e4 
 MSAFD-Tcpip [RAW/IPv6] : 2 : 23 : 3 : 0 : 0x20609 : 0xc : %SystemRoot%\system32\mswsock.dll :  : f9eab0c0-26d4-11d0-bbbf-00aa006c34e4 
 AF_UNIX : 2 : 1 : 1 : 0 : 0x20026 : 0x8 : %SystemRoot%\system32\mswsock.dll :  : a00943d9-9c2e-4633-9b59-0057a3160994 
 RSVP-TCPv6-Dienstanbieter : 2 : 23 : 1 : 6 : 0x22066 : 0x8 : %SystemRoot%\system32\mswsock.dll :  : 9d60a9e0-337a-11d0-bd88-0000c082e69a 
 RSVP-TCP-Dienstanbieter : 2 : 2 : 1 : 6 : 0x22066 : 0x8 : %SystemRoot%\system32\mswsock.dll :  : 9d60a9e0-337a-11d0-bd88-0000c082e69a 
 RSVP-UDPv6-Dienstanbieter : 2 : 23 : 2 : 17 : 0x22609 : 0x8 : %SystemRoot%\system32\mswsock.dll :  : 9d60a9e0-337a-11d0-bd88-0000c082e69a 
 RSVP-UDP-Dienstanbieter : 2 : 2 : 2 : 17 : 0x22609 : 0x8 : %SystemRoot%\system32\mswsock.dll :  : 9d60a9e0-337a-11d0-bd88-0000c082e69a 
 Hyper-V RAW : 2 : 34 : 1 : 1 : 0x20026 : 0x8 : %SystemRoot%\system32\mswsock.dll :  : 1234191b-4bf7-4ca7-86e0-dfd7c32b5445 
 MSAFD L2CAP [Bluetooth] : 2 : 32 : 1 : 256 : 0x20006 : 0x8 : %SystemRoot%\system32\mswsock.dll :  : 9fc48064-7298-43e4-b7bd-181f2089792a 
 MSAFD RfComm [Bluetooth] : 2 : 32 : 1 : 3 : 0x20026 : 0x8 : %SystemRoot%\system32\mswsock.dll :  : 9fc48064-7298-43e4-b7bd-181f2089792a
useragent_locale: de
```

so it seems like something (mswsock.dll) is blocking / crashing

**Comment by djipco** [comment]
Since both Firefox and Chrome are crashing, it looks as though it might be a driver or OS problem. It is hard to say without a deeper look. I'm fairly sure it's not a WEBMIDI.js issue, though.

You could start by reporting the issue on [Bugzilla](https://bugzilla.mozilla.org/enter_bug.cgi?product=Firefox) (Firefox) or on [bugs.chromium.org](https://bugs.chromium.org/p/chromium/) (Chrome) and work your way from there...

**Reply by schoko11** [reply]
yep, i start first with the CME support, let's see, thanks for the support!
